### DISCUSSION TITLE
How does a Noun seed become an image?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#399] [Solidity / smart contracts] [answered]
How does a Noun seed become an image?

**Comment by solimander** [comment, marked as answer]
The seed consists of array indexes that are used to access run-length encoded (RLE) image data. This RLE data is [merged and converted into an svg](https://github.com/nounsDAO/nouns-monorepo/blob/master/packages/nouns-contracts/contracts/libs/MultiPartRLEToSVG.sol#L47), then turned into a base64 data URI. OpenSea parses the data URI and displays the image.
